src/transport/SessionManager.cpp

```diff
@@ -461,6 +461,18 @@ bool SessionManager::MarkSessionsAsDefunct(const ScopedNodeId & node, const Opti
     return found;
 }
 
+void SessionManager::UpdateAllSessionsPeerAddress(const ScopedNodeId & node, const Transport::PeerAddress & addr)
+{
+    mSecureSessions.ForEachSession([&node, &addr](auto session) {
+        if (session->IsActiveSession() && session->GetPeer() == node &&
```

Comment:
We want to update even if the session is not active, right?  In particular, we want to update defunct sessions.

Reply:
What benefit to we get from updating a defunct session. I thought no outgoing messages go out on a defunct session and that only an incoming message might resurrect the defunct session

Reply:
>  I thought no outgoing messages go out on a defunct session

That's not correct.  No new outgoing exchanges can be created on a defunct session.  But if we get here, we may in fact have an exchange on that session (the one we failed first delivery for), and that we are going to try to do an MRP retransmit on that session soon, right?

The key is that the session can get marked defunct after we do our send but before the first MRP retry interval completes....